v2/spiffeid/id.go

```diff
@@ -86,32 +93,32 @@ func FromURI(uri *url.URL) (ID, error) {
 
 // ID is a SPIFFE ID
 type ID struct {
-	td   TrustDomain
-	path string
+	id      string
+	pathidx int
```

Comment:
nit: It may be useful to add a comment about the motivation of the need for this, indicating that we need to track where the path begins.

Reply:
Cool, fixed!